modules/openapi-generator/src/main/resources/rust-axum/server-operation.mustache

```diff
@@ -89,7 +89,7 @@ where
     // Header parameters
     let header_params = {
   {{/-first}}
-                let header_{{{paramName}}} = headers.get(HeaderName::from_static("{{{baseName}}}"));
+                let header_{{{paramName}}} = headers.get(HeaderName::from_static("{{{nameInLowerCase}}}"));
```

Comment:
Using nameInLowerCase instead!

Reply:
thanks for pinging me. let me take another look tomorrow

Reply:
what if the header parameter name is all upper case (e.g. X-API-CREDITS)?

understood that header are not case sensitive but shouldn't `baseName` stores the original header parameter name?

Reply:
The issue here is the map [defined here](https://docs.rs/http/1.1.0/src/http/header/name.rs.html#1006) that is used to build a `HeaderName` value. The user may define a header name using uppercase characters, something that is allowed by the OAS, and would often be fine, but in the `http` crate this would lead to a panic, so we must use the header name in lower case to prevent this. It's a necessary fix.

Reply:
understood. thanks for the explanation.